### DISCUSSION TITLE
Drag & drop (while trying to keep my hair)

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#852] [Q&A]
Hi to all

I group a few questions here regarding drag & drop (all help is welcome 👍 )

1. Copy instead of move

Is there a way to copy items from a collection to another, rather than moving them?

I saw an option in dragula `(options.copy)`: is there a way to use it?

I found a way to do it in this [codepen](https://codepen.io/galinhalx/pen/dydvabx), or even [here](https://codepen.io/galinhalx/pen/wvyJNge) based on @DmitrySharabin code,
forcing both collections to accept each other through `mv-accepts`, but

- If I remove `mv-accepts` of one collection, than mavo _moves_ the item, instead of _copy_ (normal behaviour)
- My code doesn't correspond to my need where a need collection Members to accept Persons only (and not collection Persons to accept Members) (unexpected behaviour)

2. handle on element

We already can customize buttons with` mv-actions` to add, delete, or even add to another collection (super nice features)

But can we customize it to manage handle (meaning, where to attach the drag)? 

In same previous [codepen](https://codepen.io/galinhalx/pen/dydvabx), I used `class="mv-drag-handle"` on `mv-list-item` ,
but it seems to not always work: Is this te correct way to use it?

3. Vertical transit

During drag (or gu.transit), is there a way to constrain the movement vertically inside the same collection? 

You can see it on dragula's demo n°3 [here](https://www.cssscript.com/demo/pure-javascript-drag-and-drop-library-dragula-js/)

4. Forecast of dragula bugs

I remember @LeaVerou was working on Dragula 2 #820: is version 0.2.4 of mavo already using this new release?

**Comment by LeaVerou** [comment]
> Is there a way to copy items from a collection to another, rather than moving them?

Oh good idea, we could implement this when someone drags and holds down Cmd/Ctrl.
Until then, you can duplicate an item by clicking the + while holding down Cmd/Ctrl, and then drag it like any other.

> I remember @LeaVerou was working on Dragula 2 https://github.com/mavoweb/mavo/discussions/820: is version 0.2.4 of mavo already using this new release?

Unfortunately I have not worked on that for a while. Bad timing: by the time @[bevacqua](https://github.com/bevacqua) added me as a maintainer, I had gotten swamped with other things. I still plan to work on it in the future, but it won't be very soon.

**Reply by GalinhaLX** [reply]
> > Is there a way to copy items from a collection to another, rather than moving them?
> 
> Oh good idea, we could implement this when someone drags and holds down Cmd/Ctrl. 
> 

I was more thinking about specifying a hard attribute to the list itself
(the same way we have `mv-accepts`, something like `mv-drag-options="copy"` or `mv-drag-options="move"` : exactly like Dragula is using behind)

Or better... if there's a way to use all options available by Dragula itself! by passing values through a mavo attribute

> Until then, you can duplicate an item by clicking the + while holding down Cmd/Ctrl, and then drag it like any other.

I've been able to make it work once 🥇  on...30 atempts ... is someone able to beat this score? :)


> > I remember @LeaVerou was working on Dragula 2 #820: is version 0.2.4 of mavo already using this new release?
> 
> Unfortunately I have not worked on that for a while. Bad timing: by the time @[bevacqua](https://github.com/bevacqua) added me as a maintainer, I had gotten swamped with other things. I still plan to work on it in the future, but it won't be very soon.

Power of mavo is also about multiple apps, and I guess lot of us will use movements between collections: it's part of modern UI
Will definitely vote to have this feature. I would help if my javascript level would be better.

Many thanks for your help and this amazing project @LeaVerou
